lib/App/Asciio/Elements.pm

```diff
@@ -572,50 +572,20 @@ delete $self->{SELECTION_INDEX} unless $self->get_selected_elements(1) ;
 
 #-----------------------------------------------------------------------------
 
-sub select_elements_by_search_words
+sub select_all_elements_by_search_words
 {
-my ($self, $search_words, @elements) = @_ ;
-
-my %groups_to_select ;
-
-my @filter_elements = ();
-
-for my $element (@elements) 
-	{
-	my @tmp_elements = ($element);
-	my $element_text = $self->transform_elements_to_ascii_buffer(@tmp_elements);
-	unless($element_text =~ m/$search_words/i)
-		{
-		push(@filter_elements, $element);
-		next;
-		}
+my ($self) = @_ ;
 
-	$element->{SELECTED} = ++$self->{SELECTION_INDEX} ;
-	
-	if(exists $element->{GROUP} && defined $element->{GROUP}[-1])
-		{
-		$groups_to_select{$element->{GROUP}[-1]}++ ;
-		}
-	}
+my $search_words = $self->display_edit_dialog("input search words", '', $self);
 
-# select groups
 for my $element (@{$self->{ELEMENTS}}) 
 	{
```

Comment:
This is a better implementation:

it's half the length and all the selection details are hidden (and probably better handled)

for my $element (@{$self->{ELEMENTS}}) 
	{
	$self->select_elements(1, $element)
		if $self->transform_elements_to_ascii_buffer($element) =~ m/$search_words/i ;
	}